python/cudf/cudf/core/column/datetime.py

```diff
@@ -329,9 +329,18 @@ def infer_format(element, **kwargs):
         raise ValueError("Unable to infer the timestamp format from the data")
 
     if len(second_part) > 1:
-        second_part = pd.core.tools.datetimes._guess_datetime_format(
-            "".join(second_part[1:]), **kwargs
-        )
+        # "Z" indicates Zulu time(widely used in aviation) - Which is
+        # UTC timezone that currently cudf only supports. Having any other
+        # unsuppported timezone will let the code fail below
+        # with a ValueError.
+        second_part.remove("Z")
+        second_part = "".join(second_part[1:])
+
+        if len(second_part) > 1:
+            # Only infer if second_part is not an empty string.
+            second_part = pd.core.tools.datetimes._guess_datetime_format(
```

Comment:
Is there anything in the public Pandas API we can replace this with?

Reply:
Doesn't seem like there is: https://github.com/pandas-dev/pandas/blob/6a8aca97a1f049763c03af8e3b8b0495d61d2586/pandas/tests/tools/test_to_datetime.py#L1768